### PR TITLE
logging: fix lockup when resizing window in Docker

### PR DESCRIPTION
**Pull request by oharboe**
fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/2675

logging tried to render the log message synchronously, which
caused a lockup as the main event loop was processed during
the reporting of an Xcb error.

With the upcoming https://github.com/The-OpenROAD-Project/OpenROAD/pull/3370 fix to rendering where rendering is
cancelled and restarted when it takes to long, it's not
too bad to wait until the GUI is idle before updating the
log message in the GUI.

### CODE REVIEW

**Comment by gadfort** [comment]
Once the threading pr @maliberty has is in I think we can get rid of this code, but right now it's needed to keep the logger responsive

**Comment by oharboe** [comment]
> Once the threading pr @maliberty has is in I think we can get rid of this code, but right now it's needed to keep the logger responsive

Note that the other time than rendering that the gui event loop is blocked is during tcl scripts, loading, notably.

**Comment by oharboe** [comment]
@maliberty OK to merge now that rendering is in a separate thread?

**Comment by maliberty** [comment]
re " it's not too bad to wait until the GUI is idle before updating the log message in the GUI."

For long running commands we still want the output to appear as the command progresses.  This PR prevents that.  I don't see this as mergeable.

**Comment by oharboe** [comment]
> re " it's not too bad to wait until the GUI is idle before updating the log message in the GUI."
> 
> For long running commands we still want the output to appear as the command progresses. This PR prevents that. I don't see this as mergeable.

I see  Any suggestions?

There really shouldnt be any long running blocking tasks on the GUI thread, but as a practical way forward could these long running tasks call the event handler?

**Comment by maliberty** [comment]
That's what's happening now when the log a message in the code you suggest removing.

**Comment by oharboe** [comment]
> That's what's happening now when the log a message in the code you suggest removing.

Yes, but this code is being called from Qt in an error handler under ill defined conditions where, obviously, it locks up.

Obviously it is not safe to call this method from anywhere under any conditions.

Logging should be safe to invoke anytime, anywhere.... Ideally.

File: src/gui/src/scriptWidget.cpp [hunk 1]
```diff
@@ -376,12 +376,6 @@ class ScriptWidget::GuiSink : public spdlog::sinks::base_sink<Mutex>
 
       widget_->addLogToOutput(formatted_msg, msg_color);
     }
-
-    // process widget event queue, if main thread will process new text,
-    // otherwise there is nothing to process from this thread.
-    if (QThread::currentThread() == widget_->thread()) {
-      QCoreApplication::sendPostedEvents(widget_);
```
Comment (oharboe) [inline src/gui/src/scriptWidget.cpp:383]: @maliberty How about if I checked for the XDG error message and skipped updates only for that message which is causing constipation?

**Comment by oharboe** [comment]
@maliberty Tested. Works. What do you say, merge and forget all about it? :-)

File: src/gui/src/scriptWidget.cpp (hunk 1 above)
Comment (maliberty) [reply src/gui/src/scriptWidget.cpp:383]: I'm ok with that.  Those messages are pretty useless anyways

**Comment by gadfort** [comment]
@oharboe we have a place already for filtering messages (https://github.com/The-OpenROAD-Project/OpenROAD/blob/8d56294a54f42e0844ddf36cf7847f371bdb792a/src/gui/src/gui.cpp#L75) where this should go. However, I worry that the filtering is overly broad and doesn't have good context about the full message or set of messages that are causing trouble.

**Comment by oharboe** [comment]
> @oharboe we have a place already for filtering messages (
> 
> https://github.com/The-OpenROAD-Project/OpenROAD/blob/8d56294a54f42e0844ddf36cf7847f371bdb792a/src/gui/src/gui.cpp#L75
> 
> ) where this should go. However, I worry that the filtering is overly broad and doesn't have good context about the full message or set of messages that are causing trouble.

I'm not filtering any messages, I'm just not updating the GUI immediately in the case where the error message contains this string. These messages are logged immediately to stderr.

The check is quite broad, but as the consequence is very slight, the GUI is updated a little bit later and there is no effect on stderr, I think this is good 'nuf to put lockup in Docker to bed.

**Review by maliberty — APPROVED** [review]

**Pull request merged**